scripts/tags/video.js

```diff
@@ -0,0 +1,143 @@
+(function() {
+  'use strict';
+  
+  /**
+   * search index in an array with a regex
+   * @param {Array} array
+   * @param {Regex} regex
+   * @param {Number} startpos
+   * @return {Number}
+   */
+  function reIndexOf(array, regex) {
+    let i = 0;
+    while (i < array.length) {
+      if (array[i].toString().match(regex)) {
+        return i;
+      }
+      i++;
+    }
+    return -1;
+  }
+
+  var rPath = new RegExp(
+    '((([A-Za-z]{3,9}:(?:\\/\\/)?)(?:[-;:&=+$,\\w]+@)?' +
+    '[A-Za-z0-9.-]+|(?:www.|[-;:&=+$,\\w]+@)[A-Za-z0-9.-]+)' +
+    '((?:\\/[\\+~%\\/.\\w-_]*)?\\??(?:[-+=&;%@.\\w_]*)' +
+    '#?(?:[.!\\/\\w]*))|^[A-Za-z0-9_\\/-]+\\.\\w{2,4})');
+  var rClass = /^[_a-zA-Z0-9-]+$/;
+  var rSize = /^\d+(?:|\.\d+)(?:px|%)?$/;
+  var rFigClass = /(^fig-\d{2,3}$)/;
+  var captionClass = 'caption';
+  var noCaptionClass = 'nocaption';
+  var clearClass = 'clear';
+  var autoplayClass = 'autoplay';
+  var loopClass = 'loop';
+  var mutedClass = 'muted';
+  var noControlsClass = 'nocontrols';
+  var figureClass = 'figure';
+  /**
+   * Video tag
+   *
+   * Syntax:
+   *     {% video [classes] videoURL [Optional Poster (Thumbnail) URL]
+   *     [Width] [Caption] %}
+   * E.g:
+   *     {% video loop http://example.com/video145.mp4
+   *     http://example.com/image.png 100% 95% "A beautiful sunrise" %}
```

Comment:
we can remove `100% 95%` or only leave one but having both is not correct because it only takes a width parameter.